src/main/java/com/powsybl/openloadflow/ac/outerloop/TransformerVoltageControlOuterLoop.java

```diff
@@ -53,15 +87,9 @@ public void initialize(AcOuterLoopContext context) {
             controllerBranch.setVoltageControlEnabled(false);
         }
 
-        // All transformer voltage control are disabled for the first equation system resolution.
-        double[] maxControlledNominalVoltage = new double[1];
-        maxControlledNominalVoltage[0] = Double.MIN_VALUE;
-        for (LfBus bus : context.getNetwork().getBuses()) {
-            if (!bus.isDisabled() && bus.isTransformerVoltageControlled()) {
-                maxControlledNominalVoltage[0] = Math.max(maxControlledNominalVoltage[0], bus.getNominalV());
-            }
-        }
-        ((ContextData) context.getData()).setMaxControlledNominalVoltage(maxControlledNominalVoltage[0]);
+        // all transformer voltage control are disabled for the first equation system resolution.
+        double maxControlledNominalVoltage = maxControlledNominalVoltageOverride > 0 ? maxControlledNominalVoltageOverride : calculateMaxControlledNominalVoltage(context);
```

Comment:
@annetill again here there auto should be -1. You should authorize 0 to keep all generators in voltage control.